R/guess_member_node.R

```diff
@@ -0,0 +1,58 @@
+#' Guess the Member Node
+#'
+#' @description Guess the member node that stores a Dataone object based on its
+#' unique identifier (\code{pid}) and coordinating node (\code{cn}).  In most cases
+#' the object is stored on the Production ("PROD") Node, however this function can
+#' search across all coordinating nodes.  If only one member node is identified this
+#' function returns the member node as a Dataone "MNode" object.  If multiple member
+#' nodes are identified a vector of nodes is printed.
+#'
+#' @param cn (character) A character vector of coordinate nodes to search.  Defaults to "PROD".
+#' Can be set to any combination of ("PROD", "STAGING", "STAGING2", "SANDBOX", "SANDBOX2", "DEV", "DEV2").
+#' @param pid (character) The Datone unique object identifier.
+#'
+#' @export
+#'
+#' @author Dominic Mullen, \email{dmullen17@@gmail.com}
+#'
+#' @examples
+#' \dontrun{
+#' The following two calls are equivalent:
+#' mn <- guess_member_node("doi:10.18739/A2G287")
+#' mn <- guess_member_node("PROD", "doi:10.18739/A2G287")
+#'
+#' Search all coordinating nodes:
+#' cn = c("PROD", "STAGING", "STAGING2", "SANDBOX", "SANDBOX2", "DEV", "DEV2"))
+#' mn <- guess_member_node(cn, "doi:10.18739/A2G287")
+#' }
+#'
+guess_member_node <- function(cn = "PROD", pid) {
+    stopifnot(is.character(cn))
+    stopifnot(all(cn %in% c("PROD", "STAGING", "STAGING2", "SANDBOX", "SANDBOX2", "DEV", "DEV2")))
+    stopifnot(is.character(pid))
+
+    results <- unlist(sapply(cn, function(cn) {
```

Comment:
I know we've done it this way a lot, but I think it would be more readable if the function was written in front of the sapply.

For example,
```
query_node <- function(cn){blah blah}
results <- unlist(sapply(cn, query_node))
```